### DISCUSSION TITLE
For Modal's backdrop, would `backdrop-filter` be more performant?

### DISCUSSION THREAD

**Discussion by tinymachine** [incluud/accessible-astro-components#14] [Ideas]
For the Modal component, I wonder if using `backdrop-filter` on the `.modal` element could be more performant than applying a `filter` to every element besides the modal?
https://github.com/markteekman/accessible-astro-components/blob/8d60a87a5e409cd54057033c54338ba447b8d0a1/Modal.astro#L108-L110

**Comment by markteekman** [comment]
Hey @tinymachine, thanks for submitting this idea and thank you for helping out with improving the Accessible Astro Components (also with the issues you submitted) 😊 It's true that there can be a performance cost the way this is build, but the filter is only applied to direct `<body>` elements. This can of course be a lot, depending on the elements in the page. In most cases though, it should not be noticeable. `backdrop-filter` would be a nice alternative, but since it is [not supported in Firefox](https://caniuse.com/?search=backdrop-filter) yet, I cannot use it. I will keep an eye on it though!

**Comment by tinymachine** [comment]
I wasn't aware Firefox still doesn't support it -- thanks for letting me know!

**Reply by markteekman** [reply]
Sure thing @tinymachine! And thank you for letting me know about `backdrop-filter`, I didn't even know you could do that. So I guess we both learned something 😁
